lib/segment/src/types.rs

```diff
@@ -899,21 +896,9 @@ impl TryFrom<GeoPointShadow> for GeoPoint {
 }
 
 pub trait PayloadContainer {
-    /// Return value from payload by path if it is present in the payload.
-    /// If value is not present in the payload, returns `None`.
-    ///
-    /// # Warning
-    ///
-    /// Absence of value and value `null` is NOT the same in this function.
-    fn get_value_opt(&self, path: &str) -> Option<MultiValue<&Value>>;
-
     /// Return value from payload by path.
-    /// If value is not present in the payload, returns empty value.
-    ///
-    /// # Warning
-    ///
-    /// Absence of value and value `null` is considered same in this function.
-    fn get_value(&self, path: &str) -> MultiValue<&Value>;
```

Comment:
The previous documentation didn't match the actual behavior. It doesn't have any treatment for `null` values. The actual difference between these methods was that `get_value_opt` distinguishes between a missing path and an expanding empty array. But since `get_value_opt` is only used in one place where the difference doesn't matter, I've removed it.

```rust
let p1 = Payload::new(json!({"foo": []}));
p1.get_value("foo[]"); // []
p1.get_value_opt("foo[]"); // Some([])

let p2 = Payload::new(json!({}));
p1.get_value("foo[]"); // []
p2.get_value_opt("foo[]"); // None
```

Reply:
I don't see an immediate problem with this change.

Do we have a test to assert this behavior? If not, would you mind to add one?

Reply:
Since `get_value` is just a wrapper around `get_value_from_json_map`, I could an additional test case for `get_value_from_json_map`.